src/development/platform-integration/ios/app-extensions.md

```diff
@@ -0,0 +1,216 @@
+{{site.alert.end}}
+
+1. Create a custom build of the Flutter engine that removes uses of
+   `sharedApplication` and corrects the path for the bundle.
+   Check out an [example from the community on GitHub][].
+
+1. Open the Flutter app project settings in Xcode to share build
+   configurations. 
+
+   1. Navigate to the **Info** tab.
+   1. Expand the **Configurations** group. 
+   1. Expand the **Debug**, **Profile**, and **Release** entries.
+   1. For each of these configurations, make sure the value in the
+       **Based on configuration file** drop-down menu for your
+       extension matches the one selected for the normal app target.
+
+    <figure class="site-figure {{include.class}}">
+        <div class="site-figure-container">
+            <img src='/assets/images/docs/development/platform-integration/app-extensions/xcode-configurations.png'
+            height='300'>
+        </div>
+    </figure>
+
+1. (Optional) Replace any storyboard files with an extension class if needed.
+    1. In the **Info.plist** file, delete the **NSExtensionMainStoryboard** property.
+    1. Add the **NSExtensionPrincipalClass** property.
+    1. Set this value for this property to the name of your `ViewController`.
+        For example, in an iMessage extension you would use `MessageViewController`. 
+    
+    <figure class="site-figure {{include.class}}">
+        <div class="site-figure-container">
+            <img src='/assets/images/docs/development/platform-integration/app-extensions/extension-info.png'
+            height='300'>
+        </div>
+    </figure>
+
+
+1. Embed the `FlutterViewController` as described in
+   [Adding a Flutter Screen][]. For example, you can display a 
+   specific route in your Flutter app within an iMessage extension.
+
+    ```swift
+    //This attribute tells the compiler that this piece of Swift code can be accessed from Objective-C.
+    @objc(MessagesViewController)
+
+    class MessagesViewController: MSMessagesAppViewController {
+        override func viewDidLoad() {
+            super.viewDidLoad()
+            showFlutter()
+        }
+    
+        @objc func showFlutter() {
+            // Create a FlutterViewController with an implicit FlutterEngine
+            let flutterViewController = FlutterViewController(project: nil, initialRoute: "/ext", nibName: nil, bundle: nil)
+            present(flutterViewController, animated: true, completion: nil)
+        }
+    ```
+
+{{site.alert.important}}
+Commenting out `sharedApplication` disables
+many features in the Flutter framework.
+This hasn't been tested in App Store submissions
+and might not work for some app extensions.
+
+For example, home screen Widgets and Live Activities can't use some
+lower-level APIs needed to draw Flutter UI.
+{{site.alert.end}}
+
+## Tutorials
+For step-by-step instruction for using app extensions with your
+Flutter iOS app, check out the 
+[Adding Home Screen Widgets and Live Activities to your Flutter app][] codelab.
```

Comment:
Commenting this out as this codelab isn't yet available. Also, the general Flutter codelab link (provided below) was the wrong link.

```suggestion
[Adding Home Screen Widgets and Live Activities to your Flutter app][] codelab.
{{% endcomment %}}
```